hub/core/dataset/dataset.py

```diff
@@ -426,7 +428,8 @@ def checkout(self, address: str, create: bool = False) -> str:
         Returns:
             str: The commit_id of the dataset after checkout.
         """
-        try_flushing(self)
```

Comment:
we can remove this call as we flush during copy_metas during checkout in version_control.py if a new branch is being created, otherwise if we're checking out to an existing branch/commit, explicit flushes have been added now.